tests/context-menu.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { test, expect } from '@playwright/test';
+import { waitForCustomEvent } from './lib/helper.js';
+
+test.beforeEach(async ({ page }) => {
+  await page.goto('/playground/context-menu-test.html');
+});
+
+test('Renders labels, dividers, and icons', async ({ page }) => {
+  const contextMenu = page.getByTestId('context-menu-default');
+  await expect(contextMenu).toBeVisible();
+  await expect(contextMenu.getByText('label 1')).toBeVisible();
+  await expect(contextMenu.getByTestId('context-menu-divider')).toBeVisible();
+  await expect(contextMenu.getByText('label 2')).toBeVisible();
+
+  // renders with icon
+  await expect(contextMenu.getByText('label 3')).toBeVisible();
+  await expect(
+    contextMenu.getByTestId('context-menu-icon').locator('i')
+  ).toBeVisible();
+  await expect(
+    contextMenu.getByTestId('context-menu-icon').locator('i')
+  ).toHaveClass(/icon-trash/u);
+  await expect(
+    contextMenu.locator('i:left-of(:text("label 3"))').first()
```

Comment:
Not super familiar with the locator syntax, but could this be `i.icon-trash:left-of...` or similar to collapse these two assertions?

Reply:
yup, this worked!